develop/blockchains/maintenance/storage-migrations.md

```diff
@@ -1,4 +1,183 @@
+
+`VersionedMigration` requires five type parameters:
+
+- `From` - the source version for the upgrade
+- `To` - the target version for the upgrade
+- `Inner` - the `UncheckedOnRuntimeUpgrade` implementation
+- `Pallet` - the pallet being upgraded
+- `Weight` - the runtime's [`RuntimeDbWeight`](https://paritytech.github.io/polkadot-sdk/master/frame_support/weights/struct.RuntimeDbWeight.html){target=\_blank} implementation
+
+```rust
+--8<-- 'https://raw.githubusercontent.com/paritytech/polkadot-sdk/refs/tags/polkadot-stable2409-1/substrate/frame/examples/single-block-migrations/src/migrations/v1.rs:0:122'
+```
+
+### Migration Organization
+
+Best practices recommend organizing migrations in a separate module within your pallet. Here's the recommended file structure:
+
+```plain
+my-pallet/
+├── src/
+│   ├── lib.rs       # Main pallet implementation
+│   └── migrations/  # All migration-related code
+│       ├── mod.rs   # Migrations module definition
+│       ├── v1.rs    # V0 -> V1 migration
+│       └── v2.rs    # V1 -> V2 migration
+└── Cargo.toml
+```
+
+This structure provides several benefits:
+
+- Separates migration logic from core pallet functionality
+- Makes migrations easier to test and maintain
+- Provides clear versioning of storage changes
+- Simplifies the addition of future migrations
+
+### Scheduling Migrations
+
+To execute migrations during a runtime upgrade, you need to configure them in your runtime's `Executive` pallet. Add your migrations in `runtime/src/lib.rs`:
+
+```rust
+--8<-- 'code/develop/blockchains/maintenance/storage-migrations/executive.rs'
+```
+
+### Single Block Migrations
+
+Single block migrations execute their entire logic within one block immediately following a runtime upgrade. They run as part of the runtime upgrade process through the `OnRuntimeUpgrade` trait implementation and must complete before any other runtime logic executes.
+While single block migrations are straightforward to implement and provide immediate data transformation, they carry significant risks. The most critical consideration is that they *must* complete within one block's weight limits. This is especially crucial for parachains, where exceeding block weight limits will brick the chain.
+Use single block migrations only when you can guarantee:
+
+- The migration has bounded execution time
+- Weight calculations are thoroughly tested
+- Total weight will never exceed block limits
+
+For a complete implementation example of a single-block migration, refer to the [single block migration example]( https://paritytech.github.io/polkadot-sdk/master/pallet_example_single_block_migrations/index.html){target=\_blank} in the Polkadot SDK documentation.
+
+### Multi Block Migrations
```

Comment:
niceee, liked it 🚀

Reply:
wdyt about moving "single block and multi block migrations" to be ## instead of ###

Or maybe they can be included in "## Migrations types" or whatever...